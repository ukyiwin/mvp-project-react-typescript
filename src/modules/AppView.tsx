import * as React from 'react';
import { Route, Switch, RouteComponentProps } from 'react-router-dom';
import IndexContainer from './index/IndexContainer';
import PageNotFound from '../Containers/NotFound';

export type IAppViewProps = RouteComponentProps<undefined>;

const AppView: React.StatelessComponent<IAppViewProps> = () =>
    // tslint:disable-next-line:jsx-wrap-multiline
    <div className="app-base">
        <Switch>
            <Route path="/" exact component={IndexContainer} />
            <Route component={PageNotFound} />
        </Switch>
    </div>;

export default AppView;
