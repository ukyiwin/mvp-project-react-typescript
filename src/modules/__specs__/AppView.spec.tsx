import * as React from 'react';
import { shallow } from 'enzyme';
import createHistory from 'history/createBrowserHistory';
import AppView from '../AppView';

describe('AppView', () => {
    const index = (
        <AppView
            match={{ params: undefined, isExact: true, path: '', url: '' }}
            location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
            history={createHistory()}
        />
    );
    const notFound = (
        <AppView
            match={{ params: undefined, isExact: false, path: '', url: '' }}
            location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
            history={createHistory()}
        />
    );

    it('should render correctly', () => {
        expect(shallow(index)).toMatchSnapshot();
        expect(shallow(notFound)).toMatchSnapshot();
    });
});
