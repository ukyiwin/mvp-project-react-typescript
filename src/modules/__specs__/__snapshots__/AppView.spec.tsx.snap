// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView should render correctly 1`] = `
<div
  className="app-base"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;

exports[`AppView should render correctly 2`] = `
<div
  className="app-base"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
